src/icu/charset_detector.cr

```diff
@@ -26,13 +26,21 @@ class ICU::CharsetDetector
   @@detectable_charsets : Array(String)?
 
   def initialize
+    @finalized = false
     ustatus = LibICU::UErrorCode::UZeroError
     @csdet = LibICU.ucsdet_open(pointerof(ustatus))
-    ICU.check_error!(ustatus)
+    ICU.check_error!(ustatus) { free }
```

Comment:
I'm not sure free should be called here, [the doc](http://userguide.icu-project.org/dev/codingguidelines#TOC-Functions-for-Constructors-and-Destructors) specifies:

> Each successful call to a umod_open() returns a pointer to an object that must be released by the user/owner by calling the matching umod_close().

So for me if the call is not successful, nothing should be freed.